internal/controller/connector_controller.go

```diff
@@ -0,0 +1,299 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+
+	hapi "github.com/hstreamdb/hstream-operator/api/v1alpha2"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	appsv1beta1 "github.com/hstreamdb/hstream-operator/api/v1beta1"
+)
+
+// ConnectorReconciler reconciles a Connector object
+type ConnectorReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=apps.hstream.io,resources=connectors,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=apps.hstream.io,resources=connectors/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=apps.hstream.io,resources=connectors/finalizers,verbs=update
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=deployments,verbs=create;update;patch;delete
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the Connector object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.16.0/pkg/reconcile
+func (r *ConnectorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log = logf.FromContext(ctx)
+
+	var connector appsv1beta1.Connector
+	if err := r.Get(ctx, req.NamespacedName, &connector); err != nil {
+		if !k8sErrors.IsNotFound(err) {
+			log.Error(err, "fail to fetch Connector")
```

Comment:
```suggestion
			return ctrl.Result{}, err
```

Reply:
If enable Debug log, the error will print, so we don't need print log in our code

Reply:
Got it. I will remove these logs.

Reply:
https://github.com/hstreamdb/hstream-operator/pull/59/commits/835105889745b01cfaa77799037d2a61edb12e6e Removed.

Reply:
Still have many `log.Error(); return ctrl.Result{}, err ` in other code

Reply:
Should these logs be deleted as well? They all represent important step errors. I notice hstreamdb sub-controllers will return a `requeue` struct when encountering some errors, which also log errors. I also referred to the example given in the kubebuilder documentation, which also doesn't remove all the logs.

I was beginning to think it was because you didn't think the log made by the `Get` operation was that important. So I removed them. But what if the debug log is not enabled? I guess we're also supposed to print out errors normally, right?

Reply:
Refer to https://github.com/kubernetes-sigs/controller-runtime/blob/main/TMP-LOGGING.md#logging-errors:

> Errors returned by the Reconcile implementation of the Reconciler interface are commonly logged as a Reconciler error. It's a developer choice to create an additional error log in the Reconcile implementation so a more specific file name and line for the error are returned.

I finally decided to remove all error logs for non-subresources. Since the subresources are not directly operated by the user, I think it is necessary to additionally print out a more specific reason for the error.

Reply:
Great，you are right！